lib/xlat_tables_v2/xlat_tables_internal.c

```diff
@@ -955,7 +955,7 @@ static void xlat_tables_print_internal(const uintptr_t table_base_va,
 		uint64_t *const table_base, const int table_entries,
 		const int level, const uint64_t execute_never_mask)
 {
-	assert(level <= XLAT_TABLE_LEVEL_MAX);
+	assert(level <= (int)XLAT_TABLE_LEVEL_MAX);
```

Comment:
A better fix would be change level to be a `const unsigned int`. However, this might result in more "knock-on" changes, e.g. changing `xlat_ctx_t.base_table_entries` to `unsigned int`.